src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -125,21 +125,18 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
                     notificationManager: notificationManager, queueCancellationToken: queueCancellationToken);
             }
 
-            // Go through each registered workspace, find the solution that contains the document that
-            // this request is for, and then updates it based on the state of the world as we know it, based on the
-            // text content in the document change tracker.
-
+            Solution? workspaceSolution;
             Document? document = null;
-            var workspaceSolution = await lspWorkspaceManager.TryGetHostLspSolutionAsync(requestCancellationToken).ConfigureAwait(false);
             if (textDocument is not null)
             {
-                // we were given a request associated with a document.  Find the corresponding roslyn
-                // document for this.  If we can't, we cannot proceed.
+                // we were given a request associated with a document.  Find the corresponding roslyn document for this. 
+                // There are certain cases where we may be asked for a document that does not exist (for example a document is removed)
+                // For example, document pull diagnostics can ask us after removal to clear diagnostics for a document.
                 document = await lspWorkspaceManager.GetLspDocumentAsync(textDocument, requestCancellationToken).ConfigureAwait(false);
```

Comment:
i still hate thsi :)  my preference would be that *only* those specific requests be whitelisted to refer to a document that may not be there.  that way we can throw if we don't find the document..